lib/model-utils.js

```diff
@@ -551,3 +568,15 @@ ModelUtils._coerce = function(where, options) {
   return where;
 };
 
+/**
+* A utility function which checks for nested property definitions
+*
+* @param {*} propType Property type metadata
+*
+*/
+function isNestedModel(propType) {
+  if (!propType) return false;
+  if (Array.isArray(propType)) return isNestedModel(propType[0]);
+  return propType.definition && propType.definition.properties;
```

Comment:
Here an actual boolean could be returned to keep function signature consistent. ex:
`return !!(propType.definition && propType.definition.properties;` or 
`return propType.hasOwnProperty('definition') && propType.definition.hasOwnProperty('properties');`